### DISCUSSION TITLE
Dispatched browser event runs before content is re-rendered?

### DISCUSSION THREAD

**Discussion by GregT2207** [livewire/livewire#4813] [1. Help] [answered]
It seems that when I receive an event in Alpine that's been dispatched from Livewire using ->dispatchbrowserevent(), it runs on the old verson of the DOM, not the newly rendered version.
Is this correct and if so what's the best way to dispatch an event after the new content is rendered?

**Comment by silnarm** [comment]
The `handleResponse()` function in Livewire's JS morphs the DOM before dispatching events, so this shouldn't be the case.
If you believe otherwise, could you provide a minimal example that demonstrates?

**Comment by joshhanley** [comment]
@GregT2207 yeah @silnarm is correct, the browser event should be fired after the DOM has been morphed. 
Could you share your code which reproduces the issue with as little code as possible so we can investigate it? Thanks!

**Reply by dspaccapeli** [reply]
@joshhanley is it still the case?

I found this thread while encountering a similar problem to [[discussion]](https://github.com/livewire/livewire/discussions/8578#discussion-6842791)

I tried reading the code, but I am new to the codebase and couldn't yet understand the flow.
`$wire.on` and `$wire.watch` events seem to get triggered before the DOM updates.

Example for `$wire.watch`

### Component code
```
use Livewire\Volt\Component;
use Illuminate\Support\Str;


new class extends Component {
    public $variable = '';

    public function update()
    {
        $this->variable = Str::random(10);
    }
};
```

### View code
```
<div>
    @if($variable)
        <div id="find-me">
            {{ $variable }} 
        </div>
    @endif
    <button wire:click="update" class="btn btn-primary mt-4">Update Variable</button>
</div>

@script
    <script>
        $wire.watch('variable', () =>{
            element = document.getElementById('find-me');
            console.log('element ', element.textContent);
            // null, element not found the first time you call it and with old values later on
        });
    </script>
@endscript
```

**Comment by GregT2207** [comment, marked as answer]
So after investigation I realised the DOM wasn't morphed at the point my browser event was handled because; the changes reflected in the DOM actually happened on another component called with $this->emit. It seems that emit hadn't finished when the current component's event reached the browser.

Simple fix; dispatch the browser event from the component emitted to.
